pkg/controller/dnszone/dnszone_controller.go

```diff
@@ -273,18 +282,16 @@ func (r *ReconcileDNSZone) reconcileDNSProvider(actuator Actuator, dnsZone *hive
 	if dnsZone.DeletionTimestamp != nil {
 		if zoneFound {
 			r.logger.Debug("DNSZone resource is deleted, deleting hosted zone")
-			err := actuator.Delete()
+			err = actuator.Delete()
```

Comment:
I noticed a dodgy return down on line 298 that returns err, but here we return if err isn't nil. It looked like something that could mask the real error from above because we reuse err here.

Reply:
Ah, I see.

Yeah, it looks like the `err` on L298 can only be `nil`. If I'm lining up scopes properly, it's the one from L276, and if that's not `nil` we return right away on L279.

Probably worth making that explicit (`298s/err/nil/`) so future-we don't have to go searching around like that -- but that's not related to this PR.